clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -65,11 +70,29 @@ public RouterBackedSchemaReader(
       Optional<Schema> readerSchema,
       Optional<Predicate<Schema>> preferredSchemaFilter,
       ICProvider icProvider) {
+    this(clientSupplier, readerSchema, preferredSchemaFilter, DEFAULT_SCHEMA_REFRESH_PERIOD_SEC, icProvider);
+  }
+
+  public RouterBackedSchemaReader(
+      Supplier<AbstractAvroStoreClient> clientSupplier,
+      Optional<Schema> readerSchema,
+      Optional<Predicate<Schema>> preferredSchemaFilter,
+      long valueSchemaRefreshPeriodInSec) {
+    this(clientSupplier, readerSchema, preferredSchemaFilter, valueSchemaRefreshPeriodInSec, null);
+  }
+
+  public RouterBackedSchemaReader(
+      Supplier<AbstractAvroStoreClient> clientSupplier,
+      Optional<Schema> readerSchema,
+      Optional<Predicate<Schema>> preferredSchemaFilter,
+      long valueSchemaRefreshPeriodInSec,
+      ICProvider icProvider) {
     this.storeClient = clientSupplier.get();
     this.storeName = this.storeClient.getStoreName();
     this.readerSchema = readerSchema;
-    this.preferredSchemaFilter = preferredSchemaFilter.orElse(schema -> true);
```

Comment:
Just realized that this can be fixed by changing `true` -> `false`, because it essentially disables preferred filter functionality.

Reply:
Yeah, it does. But it's quite confusing as per the naming. In my understanding, `false` just works because of the way the code is implemented.

I'll modify the handling to choose a superset schema if it is in the preferred schema list